zapcore/sampler.go

```diff
@@ -42,8 +42,13 @@ func newCounters() *counters {
 	return &counters{}
 }
 
+var nullCounter counter
+
 func (cs *counters) get(lvl Level, key string) *counter {
 	i := lvl - _minLevel
+	if i < 0 {
```

Comment:
Though it's unlikely, checking the upper bound would be helpful here too
```suggestion
	if i < 0 || i >= len(cs) {
```

Reply:
ACK